### PR TITLE
zephyr: add WASI socket support 

### PR DESCRIPTION
**Pull request by lucasAbadFr**
To address #3311

### CODE REVIEW

File: core/iwasm/common/wasm_runtime_common.c [hunk 1]
```diff
@@ -3216,7 +3216,7 @@ wasm_runtime_init_wasi(WASMModuleInstanceCommon *module_inst,
         goto fail;
     }
     addr_pool_inited = true;
-
+#if !defined(BH_PLATFORM_ZEPHYR)
```
Comment (wenyongh) [inline core/iwasm/common/wasm_runtime_common.c:3219]: Could you implement os_convert_stdin_handle, os_convert_stdout_handle, os_convert_stderr_handle and some other APIs if needed? It is strange that these critical codes are commented for zephy.

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/locking.h [hunk 2]
```diff
@@ -196,8 +196,13 @@ static inline bool
 cond_timedwait(struct cond *cond, struct mutex *lock, uint64_t timeout,
                bool abstime) REQUIRES_EXCLUSIVE(*lock) NO_LOCK_ANALYSIS
 {
+#if defined(BH_PLATFORM_ZEPHYR) 
+    return false;
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/locking.h:200]: Will you implement it in the future? How about:
```C
#if defined(BH_PLATFORM_ZEPHYR) 
    return false;
#endif
```

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c [hunk 3]
```diff
@@ -22,6 +22,7 @@
 #include "refcount.h"
 #include "rights.h"
 #include "str.h"
+#include "assert.h"
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c:25]: Seems no assert function is called? And no need to include "assert.h", just use `bh_assert` if needed.

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c [hunk 4]
```diff
@@ -647,7 +648,7 @@ fd_table_insert(wasm_exec_env_t exec_env, struct fd_table *ft,
         fd_object_release(exec_env, fo);
         return convert_errno(errno);
     }
-
+
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c:651]: Had better not add trailing spaces, or code guideline check CI will report error.

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c [hunk 5]
```diff
@@ -778,6 +779,7 @@ fd_object_get_locked(struct fd_object **fo, struct fd_table *ft, __wasi_fd_t fd,
                      __wasi_rights_t rights_inheriting)
     TRYLOCKS_EXCLUSIVE(0, (*fo)->refcount) REQUIRES_EXCLUSIVE(ft->lock)
 {
+    printf("[SSP] fd_object_get_locked\n");
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c:782]: These printf will be removed in the future, right?
And had better use os_printf or LOG_XXX instead if needed, printf may be not supported in some platforms.

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c [hunk 6]
```diff
@@ -2531,19 +2533,45 @@ wasi_ssp_sock_connect(wasm_exec_env_t exec_env, struct fd_table *curfds,
...
+    printf("[SSP] raw addr port %u\n", ntohs(*adjusted_port));
+
+    uint16_t adjusted_port_copy = ntohs(*adjusted_port);
+    
+    // adjust addr
+    addr->addr.ip4.addr.n0 = adjusted_addr_copy[3];
+    addr->addr.ip4.addr.n1 = adjusted_addr_copy[2]; 
+    addr->addr.ip4.addr.n2 = adjusted_addr_copy[1];
+    addr->addr.ip4.addr.n3 = adjusted_addr_copy[0];
+    addr->addr.ip4.port = adjusted_port_copy;
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c:2557]: Not sure why need to adjust the ip addr and port, is it possible to convert them before calling these functions? If needed, had better wrap a function and call it instead, and also control the code with zephyr macro?

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c [hunk 7]
```diff
@@ -2808,6 +2836,28 @@ wasmtime_ssp_sock_recv_from(wasm_exec_env_t exec_env, struct fd_table *curfds,
...
+    printf("[SSP] raw addr port %u\n", ntohs(*adjusted_port));
+
+    uint16_t adjusted_port_copy = ntohs(*adjusted_port);
+    
+    // adjust addr
+    src_addr->addr.ip4.addr.n0 = adjusted_addr_copy[3];
+    src_addr->addr.ip4.addr.n1 = adjusted_addr_copy[2]; 
+    src_addr->addr.ip4.addr.n2 = adjusted_addr_copy[1];
+    src_addr->addr.ip4.addr.n3 = adjusted_addr_copy[0];
+    src_addr->addr.ip4.port = adjusted_port_copy;
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c:2858]: Same as above.

File: core/shared/platform/zephyr/zephyr_clock.c [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#include "platform_api_extension.h"
```
Comment (wenyongh) [inline core/shared/platform/zephyr/zephyr_clock.c:1]: Had better add license header

File: core/shared/platform/zephyr/zephyr_file.c [hunk 9]
```diff
@@ -0,0 +1,300 @@
+#include "platform_api_extension.h"
```
Comment (wenyongh) [inline core/shared/platform/zephyr/zephyr_file.c:1]: Had better add license header

File: core/shared/platform/zephyr/zephyr_socket.c [hunk 10]
```diff
@@ -0,0 +1,1068 @@
+#include "platform_api_extension.h"
```
Comment (wenyongh) [inline core/shared/platform/zephyr/zephyr_socket.c:1]: Had better add license header

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c [hunk 11]
```diff
@@ -2863,7 +2915,26 @@ wasmtime_ssp_sock_send_to(wasm_exec_env_t exec_env, struct fd_table *curfds,
...
+
+    uint16_t *port = (uint16_t *)(ip_addr + 4);
+    printf("[SSP] raw addr port %u\n", ntohs(*port));
+
+    // adjust addr
+    adjusted_addr.addr.ip4.addr.n0 = ip_addr[3];
+    adjusted_addr.addr.ip4.addr.n1 = ip_addr[2];
+    adjusted_addr.addr.ip4.addr.n2 = ip_addr[1];
+    adjusted_addr.addr.ip4.addr.n3 = ip_addr[0];
+    adjusted_addr.addr.ip4.port = ntohs(*port);
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c:2935]: same as above

**Comment by lucasAbadFr** [comment]
I will close this pull request.

I actually opened another one (#3633) for socket + file system.

**Pull request closed without merging**